### DISCUSSION TITLE
Is there a way to search for localised term made with format()

### DISCUSSION THREAD

**Discussion by Edim7** [rappasoft/laravel-livewire-tables#1554] [Q&A] [answered]
This may not be a rappasoft issue per say, and its probably a little silly, but I am feeling completely lost.
I am trying to figure out if there is a way to search for a localised/translated value that's shown on screen but it isn't found in the database.

Here's some code to try and illustrate this

Let's say we have a column Type, where we have translated the column header as well as the displayed value within that column
```
Column::make(__('Type'), 'type')
                ->format(fn ($value) => __($value))
                ->sortable(),
```

This means that doing a callback in the searchable doesn't make sense, since we end up querying the database for a term that doesn't exist in the db, it is only displayed for the localised version

```
Column::make(__('Type'),'type')
                ->format(fn ($value) => __($value))
                ->searchable(function(Builder $query, $term) {
                    return $query->orWhere('invoices.type', 'like', '%' . trim($term) . '%');
                }),
```

Do you have any ideas if I can map the term to the underlying database value somehow because I think I've either done localisation wrong, or my database design didn't account for this, either way I feel like this is impossible to do 😅 and I don't know if its because I created an anti pattern or because I don't know enough about how to approach solving this issue.

**Comment by Edim7** [comment]
One thing that is possible is to create and I forgot to mention it is a custom filter but I don't know if this is all I can do

```
public function filters(): array
    {
        return [
            SelectFilter::make(__('Type'))
                ->options([
                    '' => __('All'),
                    'invoice' => __('invoice'),
                    'receipt' => __('receipt'),
                ])->filter(function (Builder $builder, string $value) {
                    if ($value) {
                        $builder->where('type', '=', $value);
                    }
                }),
        ];
    }
```

**Comment by lrljoe** [comment, marked as answer]
The filtering/searching happens at a database level, and with paginated results, it won't be possible to do anything on the frontend to filter using local items.

Using SelectFilter or MultiSelectDropdownFilter is your best bet here!

**Reply by Edim7** [reply]
I felt like that would be the case, thanks for confirming it.

**Reply by lrljoe** [reply]
No problem, the only other idea I can think of is using a Scout search approach, loading all of the data into an Algolia or similar, and then using their approach to translations.  But that's massive overkill and painful to implement for almost all applications.
